File: 3833-count-dominant-indices/3833-count-dominant-indices.c
int dominantIndices(int* nums, int n) {
    int cnt = 0;
    uint s = 0;
    for (int i = 0; i < n; ++i)
        s += nums[i];
    for (int i = 0; i < n - 1; ++i) {
        s -= nums[i];
        int l = n - i - 1;
        cnt += ((long long)nums[i] > (long long)(s / l));
    }
    return cnt;
}

File: 3833-count-dominant-indices/3833-count-dominant-indices.go
func dominantIndices(nums []int) int {
    n := len(nums)
    s := 0
    for _, num := range nums {
        s += num
    }
    c := 0
    for i := 0; i < n - 1; i++ {
        s -= nums[i]
        l := n - i - 1
        if int64(nums[i]) > int64(s / l) {
            c++
        }
    }
    return c
}